### PR TITLE
Added support for WDL dependencies in templates, along with uploading WDLs and dependency zips

### PR DESCRIPTION
**Pull request by KevinCLydon**
Added support for specifying dependencies when creating and updating templates.  Also added support for uploading wdls and dependencies as an option on top of being able to specify wdls and dependencies by uri.

### CODE REVIEW

File: src/carrot_cli/rest/templates.py [hunk 1]
```diff
@@ -82,3 +140,43 @@ def subscribe(template_id, email):
...
+
+def __process_wdl(params, files, wdl, type):
+    # If wdl is an http or gs uri, we'll add it to params
+    if wdl.startswith("http://") or wdl.startswith("https://") or wdl.startswith("gs://"):
+        params.append((f"{type}_wdl", wdl))
+    # Otherwise, assume wdl is a file, so we'll try to open it and put it in a files list
+    else:
+        try:
+            wdl_file = open(wdl, "rt")
+            files[f'{type}_wdl_file'] = (f'{type}.wdl', wdl_file)
```
Comment (jonn-smith) [inline src/carrot_cli/rest/templates.py:153]: Use the with construct here so the file is automatically closed:

```suggestion
            with open(wdl, 'rt') as wdl_file:
                files[f'{type}_wdl_file'] = (f'{type}.wdl', wdl_file)
```

File: tests/unit/template/test_template_command.py [hunk 2]
```diff
@@ -483,15 +513,21 @@ def test_create(create_data, caplog):
...
                 "example.com/she-ra_eval.wdl",
+                "--eval_wdl_dependencies",
+                "example.com/she-ra_eval_dep.zip",
             ],
             "params": [
                 "cd987859-06fe-4b1a-9e96-47d4f36bf819",
                 "New Sword of Protection template",
                 "This new template replaced the broken one",
                 "example.com/she-ra_test.wdl",
+                "example.com/she-ra_test_dep.zip",
```
Comment (jonn-smith) [inline tests/unit/template/test_template_command.py:528]: Your test examples are still excellent.  😁

File: src/carrot_cli/rest/request_handler.py [hunk 3]
```diff
@@ -189,18 +204,21 @@ def delete_map_by_ids(entity1, entity1_id, entity2, entity2_id):
...
+    """
     try:
         # Send request
         LOGGER.debug(
             "Sending %s request to %s with params %s and body %s",
             method,
             url,
             params,
-            body,
+            json,
```
[the remark this answers is not in the document]
Comment (KevinCLydon) [reply src/carrot_cli/rest/request_handler.py:219]: Yup, I can add that

File: src/carrot_cli/rest/templates.py (hunk 1 above)
Comment (KevinCLydon) [reply src/carrot_cli/rest/templates.py:153]: This won't work and I'm curious for your opinion on an alternative approach to this problem.  The purpose of this function is to add the appropriate value for the wdl to either the params list or the files dict (depending on whether it's a http or gs uri, or it's a local file), formatted for the request_handler to be able to use it (which is dictated by the python requests api).  The files dict has to actually contain the file object, so if we use `with`, it'll close the file before it makes it to the request_handler.  I have some ideas for ways to deal with this, so I'll figure out which way I want to do it and then push those changes.

Comment (KevinCLydon) [reply src/carrot_cli/rest/templates.py]: Just pushed a commit for this

File: src/carrot_cli/rest/templates.py [hunk 4]
```diff
@@ -82,3 +140,43 @@ def subscribe(template_id, email):
...
+def __process_wdl_dependencies(params, files, wdl_dependencies, type):
+    # If wdl is an http or gs uri, we'll add it to params
+    if wdl_dependencies.startswith("http://")\
+            or wdl_dependencies.startswith("https://")\
+            or wdl_dependencies.startswith("gs://"):
+        params.append((f"{type}_wdl_dependencies", wdl_dependencies))
+    # Otherwise, assume wdl_dependencies is a file, so we'll try to open it and put it in a files list
+    else:
+        try:
+            wdl_file = open(wdl_dependencies, "rb")
```
[the remark this answers is not in the document]
Comment (KevinCLydon) [reply src/carrot_cli/rest/templates.py]: Same as above

File: src/carrot_cli/rest/templates.py (hunk 1 above)
Comment (jonn-smith) [reply src/carrot_cli/rest/templates.py]: AH.  That's weird.  I think your solution is fine.  The `finally` block effectively does the same thing.

File: src/carrot_cli/rest/templates.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jonn-smith) [reply src/carrot_cli/rest/templates.py]: 👍

**Review by jonn-smith — APPROVED** [review]
Looks good.  Interesting that the library requires the full open file object, but it makes sense.

**Pull request merged**